axelor-account/src/main/java/com/axelor/apps/account/service/AccountService.java

```diff
@@ -126,44 +131,51 @@ public BigDecimal computeBalance(Account account, int balanceType) {
-          List<AnalyticRules> analyticRulesList =
-              accountAnalyticRulesRepository.findByAccounts(account);
-          if (analyticRulesList != null && analyticRulesList.isEmpty()) {
-            List<Long> accountAnalyticAccountList = new ArrayList<Long>();
-            List<Long> rulesAnalyticAccountList = new ArrayList<Long>();
-            account
-                .getAnalyticDistributionTemplate()
-                .getAnalyticDistributionLineList()
-                .forEach(
-                    analyticDistributionLine ->
-                        accountAnalyticAccountList.add(
-                            analyticDistributionLine.getAnalyticAccount().getId()));
-            analyticRulesList.forEach(
-                rules ->
-                    rules
-                        .getAnalyticAccountSet()
-                        .forEach(
-                            analyticAccount ->
-                                rulesAnalyticAccountList.add(analyticAccount.getId())));
-            for (Long analyticAccount : accountAnalyticAccountList) {
-              if (!rulesAnalyticAccountList.contains(analyticAccount)) {
-                throw new AxelorException(
-                    TraceBackRepository.CATEGORY_CONFIGURATION_ERROR,
-                    I18n.get(
-                        "The selected Analytic Distribution template contains Analytic Accounts which are not allowed on this account. Please select an appropriate template or modify the analytic coherence rule for this account."));
+        if (account.getAnalyticDistributionTemplate() != null) {
+          if (account.getAnalyticDistributionTemplate().getAnalyticDistributionLineList() == null) {
+            throw new AxelorException(
+                TraceBackRepository.CATEGORY_CONFIGURATION_ERROR,
+                I18n.get(
+                    "Please put AnalyticDistributionLines in the Analytic Distribution Template"));
+          } else {
+            List<AnalyticRules> analyticRulesList =
+                accountAnalyticRulesRepository.findByAccounts(account);
+            if (analyticRulesList != null && !analyticRulesList.isEmpty()) {
+              List<Long> accountAnalyticAccountList = new ArrayList<Long>();
+              List<Long> rulesAnalyticAccountList = new ArrayList<Long>();
+              account
+                  .getAnalyticDistributionTemplate()
+                  .getAnalyticDistributionLineList()
+                  .forEach(
+                      analyticDistributionLine ->
+                          accountAnalyticAccountList.add(
+                              analyticDistributionLine.getAnalyticAccount().getId()));
+              analyticRulesList.forEach(
+                  rules ->
+                      rules
+                          .getAnalyticAccountSet()
+                          .forEach(
+                              analyticAccount ->
+                                  rulesAnalyticAccountList.add(analyticAccount.getId())));
```

Comment:
not sure about this, maybe the method used to fetch analytic rules should directly return the authorized analytic account id list instead to gain in clarity and performance